### PR TITLE
Fix playback issues, while playing music, re-routing sound and voice instruction at the same time.

### PR DESCRIPTION
**Pull request by MaximAlien**
Remove tryMixAudio() in favor of tryDuckAudio() to prevent voice instructions and music playback issues.

Closing #3641.

### CODE REVIEW

File: Sources/MapboxNavigation/RouteVoiceController.swift [hunk 1]
```diff
@@ -62,8 +62,12 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
      */
     public init(navigationService: NavigationService, speechSynthesizer: SpeechSynthesizing? = nil, accessToken: String? = nil, host: String? = nil) {
         self.speechSynthesizer = speechSynthesizer ?? MultiplexedSpeechSynthesizer(accessToken: accessToken, host: host)
+        rerouteSoundPlayer = try! AVAudioPlayer(data: NSDataAsset(name: "reroute-sound", bundle: .mapboxNavigation)!.data,
```
Comment (bamx23) [inline Sources/MapboxNavigation/RouteVoiceController.swift:65]: Why does it double the existing initialization below?

Comment (MaximAlien) [reply Sources/MapboxNavigation/RouteVoiceController.swift:65]: It's not needed actually, I'd like to have it in initializer only. Updated code.

File: Sources/MapboxNavigation/RouteVoiceController.swift [hunk 2]
```diff
@@ -107,23 +111,6 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
...
-            try tryMixAudio()
-        } catch {
-            let wrapped = SpeechError.unableToControlAudio(instruction: instruction, action: .mix, underlying: error)
-            failure(wrapped)
-            return
-        }
-    }
-    
-    func tryMixAudio() throws {
-        guard speechSynthesizer.managesAudioSession else { return }
```
Comment (S2Ler) [inline Sources/MapboxNavigation/RouteVoiceController.swift:121]: This should be taken into account or new just break apps audio session without anyway to interfere.

**Review by 1ec5 — APPROVED** [review]
Remember to add a changelog entry describing the scenarios where the ducking/mixing issues are expected to improve.

**Pull request merged**